samples/test-samples.cmake

```diff
@@ -38,18 +35,31 @@ foreach (SAMPLE data-sealing)
     COMMAND ${CMAKE_COMMAND} --build ${SOURCE_DIR}/${SAMPLE}
     WORKING_DIRECTORY ${SAMPLE_BUILD_DIR})
 
-  execute_process(
-    COMMAND ${CMAKE_COMMAND} --build ${SAMPLE_BUILD_DIR} --target run
-    RESULT_VARIABLE TEST_RESULT)
+  if ((NOT DEFINED ENV{OE_SIMULATION}) OR (NOT $ENV{OE_SIMULATION}))
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} --build ${SAMPLE_BUILD_DIR} --target run
+      RESULT_VARIABLE TEST_RESULT)
+    if (TEST_RESULT)
+      message(FATAL_ERROR "Samples test '${SAMPLE}' failed!")
+    endif ()
+  endif ()
+
+  if (${SAMPLE} MATCHES "(file-encryptor|helloworld)")
+    # Of all the current samples, only file-encryptor and helloworld also work under simulation,
+    # so we test that additional scenario here.
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} --build ${SAMPLE_BUILD_DIR} --target simulate
+      RESULT_VARIABLE TEST_SIMULATE_RESULT)
+    if (TEST_SIMULATE_RESULT)
+      message(FATAL_ERROR "Samples test '${SAMPLE}' failed in simulation mode!")
+    endif ()
+  endif()
 
   # The prior common cannot succeed unless all commands before also
-  # succeeded, so testing only the result of this is sufficient.
+  # succeeded, so testing only for the results is sufficient.
   #
-  # TODO: An unfortunate side-effect of this placement is that a
-  # failed sample will cause the rest of the samples to be skipped.
-  if (TEST_RESULT)
-    message(FATAL_ERROR "Samples test '${SAMPLE}' failed!")
-  endif ()
+  # TODO: An unfortunate side-effect of using FATAL_ERROR and its placement
```

Comment:
Filed #1589.

Reply:
I added a variable to store the result if one sample failed.

Not the finest solution but it should work.